packages/db/src/models/granule-execution-history.ts

```diff
@@ -0,0 +1,50 @@
+import Knex from 'knex';
+
+import { isRecordDefined } from '../database';
+import { tableNames } from '../tables';
+
+import { PostgresGranuleExecution } from '../types/granule-execution-history';
+
+export default class GranuleExecutionHistoryPgModel {
+  readonly tableName: tableNames;
+
+  // can't extend base class because type for this data doesn't contain
```

Comment:
We could do something like this, but it will get a bit complicated with type generics so wanted to get some feedback before I go further

https://alligator.io/js/class-composition/